src/build.h

```diff
@@ -139,14 +143,17 @@ struct Plan {
 
   /// Total remaining number of wanted edges.
   int wanted_edges_;
+
+  /// Jobserver client
+  Jobserver jobserver_;
 };
 
 /// CommandRunner is an interface that wraps running the build
 /// subcommands.  This allows tests to abstract out running commands.
 /// RealCommandRunner is an implementation that actually runs commands.
 struct CommandRunner {
   virtual ~CommandRunner() {}
-  virtual size_t CanRunMore() const = 0;
+  virtual size_t CanRunMore(bool jobserver_enabled) const = 0;
```

Comment:
I suggest not changing this method's signature at all, to minimize changes.

Since the flag value is not going to change during a build, you can simply pass `plan_.JobserverEnabled()` to the `RealCommandRunner` constructor to record its value, and have `RealCommandRunner::CanRunMore()` return `SIZE_MAX` when the flag is enabled.

This should work identically, without having to change the DryCommandRunner and TestCommandRunner interfaces / implementations.

Reply:
Good point.